guardian/shortcuts.py

```diff
@@ -21,6 +21,23 @@
 from guardian.utils import get_anonymous_user, get_group_obj_perms_model, get_identity, get_user_obj_perms_model
 
 
+import django
+
+if django.VERSION >= (1, 11):
+    from django.db.models.functions import Cast
+    from django.db.models import IntegerField, CharField, AutoField
+
+
+def subquery_or_set(queryset, target_field, source_field):
+    if django.VERSION >= (1, 11):
+        if isinstance(target_field, (CharField, )):
+            return queryset.values_list(source_field)
+        elif isinstance(target_field, (AutoField, )):
+            return queryset.values_list(Cast(source_field, IntegerField()))
```

Comment:
Did you forget to add `flat=True` or did you leave it out on purpose?  (lines 34 + 36)

Reply:
No. On 34+36 lines returned `QuerySet` not list.
As I know `values('id')` and `values_list('id')` and `values_list('id', flat=True)` have same effort when resulting `QuerySet` used as subquery.

```python
In [2]: print(Company.objects.values_list('id').query)
SELECT "company_company"."id" FROM "company_company"

In [3]: print(Company.objects.values('id').query)
SELECT "company_company"."id" FROM "company_company"

In [4]: print(Company.objects.values_list('id', flat=True).query)
SELECT "company_company"."id" FROM "company_company"
```

In my practice I use `values_list('id')` for subquery usage, this is way more logical for me.

But it is not principal for me.
You want `values_list(<some field>, flat=True)`?

Reply:
> You want values_list(<some field>, flat=True)?

No, I was just wondering.